node-lib/src/options.rs

```diff
@@ -124,6 +124,13 @@ pub struct RunOptions {
     #[clap(long, value_name = "NODE")]
     pub p2p_boot_node: Option<Vec<IpOrSocketAddress>>,
 
+    /// If true, the node will assume that the nodes specified via `boot_node` won't be able
+    /// to provide it with fresh blocks; this will cause it to make a dns seed query earlier than
+    /// usual if the only nodes it knows about are those specified via `boot_node`.
+    #[clap(long)]
+    #[arg(hide = true)]
+    pub p2p_boot_nodes_will_stall: Option<bool>,
```

Comment:
I remember making a comment about this (even though I saw Gosha's comment). This option should have a better name. `force_dns_seed_query_if_none_found` is much better than this. The current name chosen is very confusing and doesn't represent the result.

Reply:
I decided to also change the logic behind the option, so that it has nothing to do with boot nodes. The option is now named `p2p_force_dns_query_if_no_global_addresses_known` and the dns query is forced if the peerdb has no global addresses at startup (where "global" addresses are those for which `is_global_unicast_ip` returns true).